tools/autograd/gen_variable_type.py

```diff
@@ -100,7 +100,7 @@
     'replication_pad1d_backward', 'replication_pad2d_backward', 'replication_pad3d_backward',
     'diag', 'masked_scatter', 'masked_select', 'index_fill', 'trace', 'polar', 'cumsum', 'rsub',
     'eig', 'lerp', 'linalg_vector_norm', 'cumprod', 'prod', 'index_copy', 'lu', 'unfold', 'unfold_backward',
-    'index', 'masked_fill', 'cross', 'lu_unpack'
+    'index', 'masked_fill', 'cross', 'lu_unpack', 'scatter', 'scatter_add'
```

Comment:
@ysiraichi If you're wondering why you needed to add this, it looks like before this PR, `at::add_scatter` was implemented as a composite op that delegated work to `at::scatter_add_`:
```
Tensor scatter_add(const Tensor & self, int64_t dim, const Tensor & index, const Tensor & source) {
    return self.clone(at::MemoryFormat::Preserve).scatter_add_(dim, index, source);
 }
```

And that includes delegating autograd. It looks like `scatter_add_` supports complex autograd, which means that `scatter_add` inherited that support too.

Now that `scatter_add` is structured, it needs its own autograd support. It looks like you added that in `derivatives.yaml`, but it also looks like the [codegen is set up](https://github.com/pytorch/pytorch/blob/12bb1e86ed150cfed4b6a7496db982ab6ce3450e/tools/autograd/gen_variable_type.py#L561) so that operators with specific return types (including `Tensor`) don't automatically get complex autograd support; you have to add them to this list once you're convinced that the implementation works with complex autograd.

Reply:
Ah, you are totally right!
I hadn't seen that `differentiable_OUTPUTS`... That makes total sense now!
Thanks for that.